connexion/cli.py

```diff
@@ -104,6 +131,29 @@ def run(spec_file,
 
     - BASE_MODULE_PATH (optional): filesystem path where the API endpoints handlers are going to be imported from.
     """
+    if wsgi_server and server:
+        raise click.BadParameter(
+            "these options are mutually excludent",
+            param_hint="'wsgi-server' and 'server'"
+        )
+    elif wsgi_server:
+        server = wsgi_server
+
+    if server is None:
+        server = DEFAULT_SERVERS[app_framework]
+
+    if app_framework not in AVAILABLE_SERVERS[server]:
+        message = "Invalid server '{}' for app-framework '{}'".format(
+            server, app_framework
+        )
+        raise click.UsageError(message)
+
+    if app_framework == AIOHTTP_APP:
```

Comment:
I did this check here because if a user input is "--app-framework=flask --server=aiohttp" the error that will raise is "invalid server for 'flask'".

If the check is done on "validate_server_requirements" the error that will raise is "aiohttp not found".

But I can move this check to the "validate_server_requirements" function if you prefer